### PR TITLE
Build multiple configurations

### PR DESCRIPTION
**Pull request by ericstj**
Replacement for https://github.com/dotnet/corefx/pull/15541

Still need to clean up commits.

/cc @weshaggard @joperezr

### CODE REVIEW

File: Tools-Override/FrameworkTargeting.targets [hunk 1]
```diff
@@ -4,10 +4,10 @@
   <ItemGroup>
     <TargetingPackDirs Include="$(RefPath)" />
     <AdditionalReferencePaths Include="@(TargetingPackDirs)" />
+    <Reference Condition="($(NuGetTargetMoniker.StartsWith('.NETStandard')) OR '$(IsReferenceAssembly)' != 'true') AND Exists('$(RefPath)netstandard.dll') AND '$(MSBuildProjectExtension)' != '.pkgproj'" Include="netstandard" />
```
Comment (weshaggard) [inline Tools-Override/FrameworkTargeting.targets]: Could we please create a property that controls this? Something like Include/ExcludeDefaultReferences and set that to true/false based on these various conditions. I really find it difficult to parse this extremely long set of conditions.

Comment (weshaggard) [inline Tools-Override/FrameworkTargeting.targets]: It also doesn't allow for individual projects to override the behavior.

Comment (ericstj) [inline Tools-Override/FrameworkTargeting.targets]: Yeah, I don't like this either, wasn't sure if there might even be a better place for this to live. (eg: targetingpacks.props.)  For now I'll just polish it up a little.

Comment (ericstj) [inline Tools-Override/FrameworkTargeting.targets]: How does this look: 
```xml
  <PropertyGroup Condition="'$(IncludeDefaultReferences)' == ''">
    <IncludeDefaultReferences Condition="'$(MSBuildProjectExtension)' == '.csproj'">true</IncludeDefaultReferences>
    <IncludeDefaultReferences Condition="'$(MSBuildProjectExtension)' == '.vbproj'">true</IncludeDefaultReferences>
  </PropertyGroup>

  <ItemGroup>
    <!-- netstandard is a default reference whenever building for NETStandard or building an implementation assembly -->
    <DefaultReferences Condition="$(NuGetTargetMoniker.StartsWith('.NETStandard') OR '$(IsReferenceAssembly)' != 'true'"
                       Include="netstandard" />

    <!-- Include default references when specified and they exist -->
    <Reference Condition="'$(IncludeDefaultReferences)' =='true' AND Exists('$(RefPath)%(Identity).dll')" Include="@(DefaultReferences)" />
  </ItemGroup>
```

Comment (weshaggard) [inline Tools-Override/FrameworkTargeting.targets]: Better but I think I would tweak it along the lines of:

```
  <PropertyGroup Condition="'$(IncludeDefaultReferences)' == ''">
    <IncludeDefaultReferences Condition="'$(MSBuildProjectExtension)' == '.csproj'">true</IncludeDefaultReferences>
    <IncludeDefaultReferences Condition="'$(MSBuildProjectExtension)' == '.vbproj'">true</IncludeDefaultReferences>
  </PropertyGroup>

  <ItemGroup Condition="'$(IncludeDefaultReferences)' =='true'" >
    <!-- netstandard is a default reference whenever building for NETStandard or building an implementation assembly -->
    <DefaultReference Condition="$(NuGetTargetMoniker.StartsWith('.NETStandard') OR '$(IsReferenceAssembly)' != 'true'"
                       Include="netstandard" />

    <!-- Include default references when specified and they exist -->
    <Reference Condition="'Exists('$(RefPath)%(Identity).dll')" Include="@(DefaultReference)" />
  </ItemGroup>
```

Also I think this is the right file as I want all of targetingpacks.props to go into here.

**Pull request merged**